sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -36,6 +36,10 @@ case class InMemoryTableScanExec(
     @transient relation: InMemoryRelation)
   extends LeafExecNode with ColumnarBatchScan {
 
+  override val nodeName: String = {
+    s"Scan in memory table ${relation.cacheBuilder.tableName.map(_.replace("`", "")).getOrElse("")}"
```

Comment:
Here, `memory table` is misleading to end users, if the cached data does not correspond to a table.